docs/styleguide/PyGuide.md

```diff
@@ -0,0 +1,820 @@
+<a id="comments-and-docstrings"></a>
+### 3.5 Comments and Docstrings 
+
+Be sure to use the right style for module, function, method docstrings and
+inline comments.
+
+<a id="s3.5.1-modules"></a>
+<a id="351-modules"></a>
+<a id="modules"></a>
+#### 3.5.1 Modules 
+
+Every file should contain license boilerplate.
+
+```python
+"""
+ Copyright (c) 2020 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+```
+
+<a id="s3.5.2-functions-and-methods"></a>
+<a id="352-functions-and-methods"></a>
+<a id="functions-and-methods"></a>
+#### 3.5.2 Functions and Methods 
+
+In this section, "function" means a method, function, or generator.
+
+A function must have a docstring, unless it meets all of the following criteria:
+-   not externally visible
+-   very short
+-   obvious
+
+```python
+def load_state(model: torch.nn.Module, state_dict_to_load: dict, is_resume: bool = False) -> int:
+    """
+    Used to load a checkpoint containing a compressed model into an NNCFNetwork object, but can
+    be used for any PyTorch module as well. Will do matching of state_dict_to_load parameters to
+    the model's state_dict parameters while discarding irrelevant prefixes added during wrapping
+    in NNCFNetwork or DataParallel/DistributedDataParallel objects, and load the matched parameters
+    from the state_dict_to_load into the model's state dict.
+    
+    :param model: The target module for the state_dict_to_load to be loaded to.
+    :param state_dict_to_load: A state dict containing the parameters to be loaded into the model.
+    :param is_resume: Determines the behavior when the function cannot do a successful parameter match
+    when loading. If True, the function will raise an exception if it cannot match the state_dict_to_load
+    parameters to the model's parameters (i.e. if some parameters required by model are missing in
+    state_dict_to_load, or if state_dict_to_load has parameters that could not be matched to model parameters,
+    or if the shape of parameters is not matching). If False, the exception won't be raised.
+    Usually is_resume is specified as False when loading uncompressed model's weights into the model with
+    compression algorithms already applied, and as True when loading a compressed model's weights into the model
+    with compression algorithms applied to evaluate the model.
+    :return: The number of state_dict_to_load entries successfully matched and loaded into model.
+    """
+```
```

Comment:
I think it should be formatted because it is very hard to read

```python
def load_state(model: torch.nn.Module, state_dict_to_load: dict, is_resume: bool = False) -> int:
    """Used to load a checkpoint containing a compressed model into an NNCFNetwork object, but can
    be used for any PyTorch module as well. Will do matching of state_dict_to_load parameters to
    the model's state_dict parameters while discarding irrelevant prefixes added during wrapping
    in NNCFNetwork or DataParallel/DistributedDataParallel objects, and load the matched parameters
    from the state_dict_to_load into the model's state dict.

    :param model: The target module for the state_dict_to_load to be loaded to.
    :param state_dict_to_load: A state dict containing the parameters to be loaded into the model.
    :param is_resume: Determines the behavior when the function cannot do a successful parameter match
        when loading. If True, the function will raise an exception if it cannot match the state_dict_to_load
        parameters to the model's parameters (i.e. if some parameters required by model are missing in
        state_dict_to_load, or if state_dict_to_load has parameters that could not be matched to model parameters,
        or if the shape of parameters is not matching). If False, the exception won't be raised.
        Usually is_resume is specified as False when loading uncompressed model's weights into the model with
        compression algorithms already applied, and as True when loading a compressed model's weights into the 
        model with compression algorithms applied to evaluate the model.
    :return: The number of state_dict_to_load entries successfully matched and loaded into model.
    """
```

Reply:
cc' @vshampor

Reply:
I've updated the formatting.